web/js-dom-methods.md

```diff
@@ -51,6 +51,32 @@ Prominent companies like Netflix, Walmart, Paypal built their entire application
 - **There are several JavaScript methods that achieve similar goals but through different means.** You should understand how they're different so you can use them appropriately.
   - `.appendChild()` vs `.insertAdjacentText()`
 
+### DOM Event Basics
+
+DOM stands for Document Object Model. Events in DOM are actions that take place in the system, the browser activates many events and provides a mechanism with the help of which we can take actions automatically when the event occurs. DOM events help in adding event handlers on elements in the HTML document with the help of JavaScript. Example: On clicking a button we can change the background color of the webpage. 
+Each available event has an event handler/event listener, that is a block of code, that runs when the event fires.
+Some common examples of events are:
+- mouse events (MouseEvent): mousedown, mouseup, click, dblclick, mousemove, mouseover, mousewheel, mouseout, contextmenu
+- window events: scroll, resize, hashchange, load, unload
+- keyboard events (KeyboardEvent): keydown, keypress, keyup
+
+### .addEventListener() Function
+
+The addEventListener() method allows us to add event listeners on the HTML DOM object such as Document, Element, Window. The method takes an event to listen for and a second argument that has to be called when the event is fired. The first argument should be a valid string. The second argument can be a function or an object that implements EventListener on the specified event on which it is called.
+
+Syntax: `target.addEventListener(type, listener [, options]);`
+
+If you no longer need a particular event listener, you can remove it using removeEventListener.
+
+### Event Bubbling and Propagation
+
+Event propagation is the flow in which the events propagate to reach its target. It is the process of calling all the listeners for the given event type. We can register several listeners on one node for the same event type. The listeners are called in the order of their registration on reaching the node. The flow of propagation can be to the event target from the window and back, thus making it bi-directional.
```

Comment:
This is pretty dry stuff.  Since this is the first time apprentices will have heard about DOM manipulation, can you please choose an element, then use it to describe event examples for each of the concepts here in your new section? A diagram would be even better.

Reply:
Okay sure, thank you.

Reply:
@alodahl I have make the changes. Please review. Thanks.